src/pip/_internal/configuration.py

```diff
@@ -283,8 +284,13 @@ def _construct_parser(self, fname):
         # Doing this is useful when modifying and saving files, where we don't
         # need to construct a parser.
         if os.path.exists(fname):
-            parser.read(fname)
-
+            try:
+                parser.read(fname)
+            except UnicodeDecodeError:
+                raise ConfigurationError((
+                    "ERROR: Configuration file contains invalid %s characters.\n"
```

Comment:
Line length error here...

Reply:
Just noticed you'd already made the point about the test :-)

Reply:
> Line length error here...

Yeps! Pushing...